### ISSUE TITLE
$elemMatch fails to find records (BUG)

### ISSUE THREAD

**Issue by canuto** [Softmotions/ejdb#208]
A query using $elemMatch fails to find records when the following are true:

1. it contains two (or more) array properties and one of them is a complete substring of the other (starting at index 0 - ex: "a" and "ab")
2. the shortest property appears first

Another bug: when we have another record similar to the one that is found, the first one is included in the result twice.

Example code to demonstrate:
```
const EJDB = require("ejdb")
const db = EJDB.open("testdb")
var notFoundObject = {
    "name": "notFoundObject",
    "a": [
        {
            "ignoreme": "hello",
        }
    ],
    "ab": [
        {
            "findme": 50
        }
    ]
}

var foundButNotIncludedObject = {
    "name": "foundButNotIncludedObject",
    "ab": [
        {
            "findme": 50
        }
    ],
    "a": [
        {
            "ignoreme": "hello",
        }
    ]
}

var foundObject = {
    "name": "foundObject",
    "ab": [
        {
            "findme": 50
        }
    ],
    "a": [
        {
            "ignoreme": "hello",
        }
    ]
}

db.dropCollection("testcoll");
db.save("testcoll", notFoundObject);
db.save("testcoll", foundObject);
db.save("testcoll", foundButNotIncludedObject);

var cursor = db.find("testcoll", { ab: { $elemMatch: { findme: 50 } } });

let results = []
while (cursor.hasNext()) {
    results.push(cursor.object());
    cursor.next();
}
// we should have three objects, but we have only two, and they are the same!
console.log("results length is", results.length);
console.log(results);
```
Output:
```
results length is 2
[
  {
    "_id": "5b282a0f8a6de12900000001",
    "name": "foundObject",
    "ab": [
      {
        "findme": 50
      }
    ],
    "a": [
      {
        "ignoreme": "hello"
      }
    ]
  },
  {
    "_id": "5b282a0f8a6de12900000001",
    "name": "foundObject",
    "ab": [
      {
        "findme": 50
      }
    ],
    "a": [
      {
        "ignoreme": "hello"
      }
    ]
  }
]
```

**Comment by adamansky** [comment]
Hi Martin! Thank you for the descriptive report. Currently `EJDB 1.x` persists in unmaintainable state, because all of my resources concentrated on `EJDB2` - a fully re-implemented database engine (See https://github.com/Softmotions/ejdb/issues/183) So I don't give you any promises concerning this issue hoping EJDB2 will be released shortly (at end of summer).

**Comment by canuto** [comment]
Ok, thank you :-)

**Comment by adamansky** [comment]
Closed in favour to ejdb 2.0 https://github.com/Softmotions/ejdb/blob/master/README.md
